### PR TITLE
[MERGE] sms, various: improve SMS integration with contact, calendar, crm, server actions

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
PURPOSE

Provide fixes to SMS behavior, notably in composer.
Better integration in various applications: contacts, calendar, crm.
Allow to automatically send SMS from server actions.

SPECIFICATIONS

See sub commits for more details.

Linked to task 1925950 and 1935280

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo retry

**Comment by tde-banana-odoo** [comment]
@robodoo r+ retry rebase-merge do something ugly son of a bot

**Comment by tde-banana-odoo** [comment]
@d-fence

**Comment by tde-banana-odoo** [comment]
@Xavier-Do

**Comment by tde-banana-odoo** [comment]
@robodoo I hate you so much

**Comment by d-fence** [comment]
@tde-banana-odoo it's merging, what is the problem ?

**Comment by tde-banana-odoo** [comment]
@d-fence Until 5 minutes ago there was no tag and no information 2 hours after a retry. 2 hours to push a retry on a falsy-positive ...

File: addons/sms/models/ir_actions.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    )
+    sms_mass_keep_log = fields.Boolean('Log a note', default=True)
+
+    @api.constrains('state', 'model_id')
+    def _check_sms_capability(self):
+        for action in self:
+            if action.state == 'sms' and not action.model_id.is_mail_thread:
+                raise ValidationError(_("Sending SMS can only be done on a mail.thread model"))
+
+    @api.model
```
Comment (mreficent) [inline addons/sms/models/ir_actions.py]: `api.model` is dead in `master`

Comment (Xavier-Do) [reply addons/sms/models/ir_actions.py]: api.multi is dead api.model is still usefull

**Pull request closed without merging**